core/src/main/java/org/apache/iceberg/rest/requests/RenameTableRequest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * A REST request to rename a table.
+ */
+public class RenameTableRequest {
+
+  private TableIdentifier source;
+  private TableIdentifier destination;
+
+  @SuppressWarnings("unused")
+  public RenameTableRequest() {
+    // Needed for Jackson Deserialization.
+  }
+
+  private RenameTableRequest(TableIdentifier source, TableIdentifier destination) {
+    this.source = source;
+    this.destination = destination;
+    validate();
+  }
+
+  public RenameTableRequest validate() {
+    Preconditions.checkArgument(source != null, "Invalid source table: null");
+    Preconditions.checkArgument(destination != null, "Invalid destination table: null");
+    Preconditions.checkArgument(source.hasNamespace(), "Invalid source table: missing namespace");
+    Preconditions.checkArgument(destination.hasNamespace(), "Invalid destination table: missing namespace");
```

Comment:
Do we want to allow for the empty namespace to be used? Its not allowed in some other catalogs, but it seems like that might be beneficial for e.g. catalogs that don't support namespaces.

Reply:
Yes, I don't see a reason not to allow requesting the empty namespace. That's valid in many catalogs.

Reply:
Ok will remove that.

Reply:
Removed.